DEV/org.openl.commons/src/org/openl/util/StringUtils.java

```diff
@@ -558,4 +556,33 @@ public static int lastNonSpace(CharSequence text, int start, int end) {
         }
         return -1;
     }
+
+    public static String normalizeSpace(String str) {
+        int size = str.length();
+        StringBuilder sb = new StringBuilder();
+        int whitespacesCount = 0;
+        boolean startWhitespaces = true;
+
+        for (int i = 0; i < size; ++i) {
+            char actualChar = str.charAt(i);
+            boolean isWhitespace = Character.isWhitespace(actualChar) || actualChar == 160;
+            if (isWhitespace) {
+                if (whitespacesCount == 0 && !startWhitespaces) {
+                    sb.append(" ");
+                }
+
+                ++whitespacesCount;
+            } else {
+                startWhitespaces = false;
+                sb.append(actualChar);
+                whitespacesCount = 0;
+            }
+        }
+
+        if (startWhitespaces) {
+            return "";
+        } else {
+            return sb.toString();
+        }
+    }
 }

```

Comment:
This algorithm consumes much more memory (approximately twice more), than the original from the original from Apache commons lang3.
Moreover it was not covered by tests and can fails with NullPointerException.